pkg/cosign/verify.go

```diff
@@ -315,13 +304,19 @@ func CheckCertificatePolicy(cert *x509.Certificate, co *CheckOpts) error {
 						break
 					}
 				}
+				if regex.MatchString(cert.Subject.String()) {
```

Comment:
Same here, a certificate subject is not the same as the subject alternative name, and we don't expect fulcio certificates to ever set the certificate subject.

Reply:
Do we care about non-Fulcio certificates that don't include SANs? We're using such a certificate in test/sign_blob_test.sh, so I figured it must be a supported use case.

Reply:
I don't think we should. The bring-your-own-pki story needs to be fleshed out and we'd need to think through how to represent a subject (do we do something like `CN=foo;O=bar`, or do we support flags for each subject component?)

Do you want me to help regenerate a test certificate?

Reply:
Yes, please.

Reply:
added comment